GitUI/Shells/BashShell.cs

```diff
@@ -18,16 +19,32 @@ public BashShell()
             Name = ShellName;
             Icon = Images.GitForWindows;
 
-            ExecutableName = BashExe;
-            if (PathUtil.TryFindShellPath(ExecutableName, out var exePath))
+            if (PathUtil.TryFindShellPath(GitBashExe, out var exePath))
             {
+                ExecutableName = GitBashExe;
                 ExecutablePath = exePath;
+
+                // Try to find bash or sh below to set ExecutableCommandLine, as git-bash.exe cannot be connected to the built-in console.
+            }
+
+            if (PathUtil.TryFindShellPath(BashExe, out exePath))
+            {
+                if (ExecutablePath is null)
+                {
+                    ExecutableName = BashExe;
+                    ExecutablePath = exePath;
+                }
+
                 ExecutableCommandLine = $"{exePath.Quote()} --login -i";
             }
             else if (PathUtil.TryFindShellPath(ShExe, out exePath))
             {
-                ExecutableName = ShExe;
-                ExecutablePath = exePath;
+                if (ExecutablePath is null)
+                {
+                    ExecutableName = BashExe;
+                    ExecutablePath = exePath;
+                }
```

Comment:
Line:30 - look for `BashExe`, line:34 - set to `BashExe`. Line:40 - look for `ShExe`, line:34 - set to `BashExe`? This at least deserves a comment explaining why this is.

Also lines:31-39 and lines:41:49 are identical, and **if** the above is correct, then, I think, this should work:
```suggestion
            if (PathUtil.TryFindShellPath(BashExe, out exePath) || PathUtil.TryFindShellPath(ShExe, out exePath))
            {
                if (ExecutablePath is null)
                {
                    ExecutableName = BashExe;
                    ExecutablePath = exePath;
                }

                ExecutableCommandLine = $"{exePath.Quote()} --login -i";
            }
```

Reply:
Good catch. It was a copy & paste mistake. (The code before #8213 did not have this duplicated code. But it's only 2.5 copies...)